### PR TITLE
116 limit event data capturing

### PR DESCRIPTION
**Pull request by rschwietzke**

### CODE REVIEW

File: config/reportgenerator.properties [hunk 1]
```diff
@@ -25,9 +25,24 @@
...
-## test case (defaults to true).
-#com.xceptance.xlt.reportgenerator.events.groupByTestCase = false
+## test case (is true by default).
+# com.xceptance.xlt.reportgenerator.events.groupByTestCase = true
+
+## How many different events per test case should be evaluated during report
+## generation? This is a safety limited because the event name should not
+## contain any dynamic data, the message should (if needed) (default 100).
+## In case we dropped any events by name, we will report an artificial warning
+## event in the report (XLT::DroppedEvents Due to Naming).
```
Comment (jowerner) [inline config/reportgenerator.properties:39]: Please review comment once more and adjust the name of the artificial event.

File: config/reportgenerator.properties [hunk 2]
```diff
@@ -25,9 +25,24 @@
...
+# com.xceptance.xlt.reportgenerator.events.eventLimit = 100
+
+## This is the limited of distinct messages per event per test case that will be
+## evaluated during report generation. All messages above that limit will be ignored
+## and only counted as dropped (default 250).
```
Comment (jowerner) [inline config/reportgenerator.properties:44]: Please review comment once more.

File: src/main/java/com/xceptance/xlt/report/providers/EventReport.java [hunk 3]
```diff
@@ -29,20 +29,121 @@
...
+        messageMap = null;
+    }
+
+    /**
+     * Add another event report
+     *
+     * @param data the data to add
+     * @return our updated instance
+     */
+    public EventReport add(final EventReport data)
```
Comment (jowerner) [inline src/main/java/com/xceptance/xlt/report/providers/EventReport.java:113]: Rename to "merge"?

File: src/main/java/com/xceptance/xlt/report/providers/EventReport.java [hunk 4]
```diff
@@ -29,20 +29,121 @@
...
+        /**
+         * We are not going to drop any messages anymore now when summing up things!!
+         */
+        data.messageMap.values().forEach(e -> {
+            var info = this.messageMap.get(e.info);
+            if (info == null)
+            {
+                info = new EventMessageInfo();
+                info.info = e.info;
+                info.count = 1;
```
Comment (jowerner) [inline src/main/java/com/xceptance/xlt/report/providers/EventReport.java:127]:
```suggestion
                info.count = e.count;
```

File: src/main/java/com/xceptance/xlt/report/providers/EventsReportProvider.java [hunk 5]
```diff
@@ -68,7 +81,67 @@ public void setConfiguration(ReportProviderConfiguration config)
...
+
+                this.testCaseToEventMap.put(eventData.getTestCaseName(), stat);
+            }
+            else
+            {
+                //
+                EventReport eventReport = stat.get(eventData.getName());
+                if (eventReport == null)
+                {
+                    if (stat.size() >= eventLimitPerTestCase)
```
Comment (jowerner) [inline src/main/java/com/xceptance/xlt/report/providers/EventsReportProvider.java:123]: Move this check top-level? Currently, if eventLimitPerTestCase is 0, we will get 1 entry nevertheless.

File: src/main/java/com/xceptance/xlt/report/providers/EventsReportProvider.java [hunk 6]
```diff
@@ -91,86 +164,65 @@ public void run()
...
-                messageInfosPerEvent.put(eventStat.getMessage(), messageInfo);
-            }
-            else
-            {
-                messageInfo.count++;
-            }
+        // in case we have eventsDropped > 0, we insert a virtual event
+        if (eventsDropped > 0)
+        {
+            final EventReport dropped = new EventReport("XLT Internal", "XLT::Dropped events due to bad naming");
```
Comment (jowerner) [inline src/main/java/com/xceptance/xlt/report/providers/EventsReportProvider.java:211]: Adjust event name?

**Comment by jowerner** [comment]
In case we don't group events by test case, the resulting list of events in the report may exceed the configured event limit. Obey the event limit when merging event data reports together.

**Comment by rschwietzke** [comment]
Addressed all comments. Please review. Also decreased the default amount of message per event to 100 from 250.

**Pull request merged**